feature-charity/src/main/java/com/hedvig/android/feature/charity/ui/CharityScreen.kt

```diff
@@ -0,0 +1,272 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Info
+import androidx.compose.material.rememberModalBottomSheetState
+import androidx.compose.material3.windowsizeclass.ExperimentalMaterial3WindowSizeClassApi
+import androidx.compose.material3.windowsizeclass.WindowSizeClass
+import androidx.compose.material3.windowsizeclass.WindowWidthSizeClass
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import coil.ImageLoader
+import coil.compose.AsyncImagePainter
+import coil.compose.rememberAsyncImagePainter
+import coil.request.ImageRequest
+import coil.size.Size
+import com.google.accompanist.placeholder.PlaceholderHighlight
+import com.google.accompanist.placeholder.material.fade
+import com.google.accompanist.placeholder.material.placeholder
+import com.hedvig.android.core.designsystem.theme.textColorLink
+import com.hedvig.android.core.ui.FullScreenHedvigProgress
+import com.hedvig.android.core.ui.appbar.TopAppBarWithBack
+import com.hedvig.android.core.ui.genericinfo.GenericErrorScreen
+import com.hedvig.android.core.ui.preview.rememberPreviewImageLoader
+import com.hedvig.android.feature.charity.CharityInformation
+import com.hedvig.android.feature.charity.CharityUiState
+import hedvig.resources.R
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+internal fun CharityScreen(
+  uiState: CharityUiState,
+  retry: () -> Unit,
+  goBack: () -> Unit,
+  imageLoader: ImageLoader,
+  windowSizeClass: WindowSizeClass,
+) {
+  val coroutineScope = rememberCoroutineScope()
+  val sheetState = rememberModalBottomSheetState(
+    initialValue = ModalBottomSheetValue.Hidden,
+    skipHalfExpanded = true,
+  )
+  ModalBottomSheetLayout(
+    sheetState = sheetState,
+    sheetContent = {
+      CharityBottomSheet(
+        isShowing = sheetState.isVisible,
+        closeSheet = { coroutineScope.launch { sheetState.hide() } },
+      )
+    },
+    modifier = Modifier.fillMaxSize(),
+  ) {
+    ScreenContent(
+      uiState = uiState,
+      showSheet = { coroutineScope.launch { sheetState.show() } },
+      goBack = goBack,
+      retry = retry,
+      imageLoader = imageLoader,
+      windowSizeClass = windowSizeClass,
+    )
+  }
+}
+
+@Composable
+private fun ScreenContent(
+  uiState: CharityUiState,
+  showSheet: () -> Unit,
+  goBack: () -> Unit,
+  retry: () -> Unit,
+  imageLoader: ImageLoader,
+  windowSizeClass: WindowSizeClass,
+) {
+  Box(Modifier.fillMaxSize()) {
+    Column {
+      TopAppBarWithBack(
+        onClick = { goBack() },
+        title = "",
```

Comment:
This line seems like an indicator that the API of `TopAppBarWithBack` could maybe be improved 🤔 

Alternatively that it's inappropriate for the usecase.

Reply:
Yeah I agree, I still like to call the `TopAppBarWithBack` function here to make sure that we've got the same look, height etc as the other screens, but the API needs to be updated to support an empty title without this "" weird looking call site